src/lib/operators/join_sort_merge.cpp

```diff
@@ -517,6 +732,34 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
 
   // Performs the join on a single cluster. Runs of entries with the same value are identified and handled together.
   // This constitutes the merge phase of the join. The output combinations of row ids are determined by _join_runs.
+  // The main logic of the algorithm is the following:
+  // The values of the left and right join columns are grouped into runs (entries with the same value). The values of
+  // the runs are compared and then we determine if we advanced the right run, left run or both runs. If the comparison
```

Comment:
I guess that should be rather "then we determine if we advance the right run, left runCOMMA, or both runs." (comma and advance in present).